zk/tx/tx.go

```diff
@@ -189,6 +189,8 @@ func DecodeBatchL2Blocks(txsData []byte, forkID uint64) ([]DecodedBatchL2Data, e
 	return result, nil
 }
 
+type TxDecoder func([]byte, byte, uint64) (types.Transaction, uint8, error)
+
```

Comment:
Consider using named parameters, for the sake of brevity.

```suggestion
type TxDecoder func(encodedTx []byte, gasPricePercentage uint8, forkID uint64) (types.Transaction, uint8, error)
```